### PR TITLE
Support old versions of React

### PR DESCRIPTION
**Pull request by rom10**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "react-jsonschema-form",
-  "version": "0.40.0",
+  "name": "book-md-react-jsonschema-form",
```
Comment (n1k0) [inline package.json]: hmm.

**Comment by n1k0** [comment]
Which version of React introduced this incompatibility?

File: package.json (hunk 1 above)
Comment (rom10) [inline package.json]: Reverted

**Comment by rom10** [comment]
@n1k0 https://github.com/facebook/react/blob/master/CHANGELOG.md#1500-april-7-2016

> Rendering null now uses comment nodes. Previously null would render to <noscript> elements. We now use comment nodes. This may cause issues if making use of :nth-child CSS selectors. While we consider this rendering behavior an implementation detail of React, it's worth noting the potential problem. (@spicyj in #5451)

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-jsonschema-form",
-  "version": "0.40.0",
+  "version": "0.40.2",
```
Comment (n1k0) [inline package.json]: This should be removed too, I'll take care of the version ;)

File: src/components/fields/DescriptionField.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ import React, {PropTypes} from "react";
 function DescriptionField(props) {
   const {id, description} = props;
   if (!description) {
-    return null;
+    return <div />;
```
Comment (n1k0) [inline src/components/fields/DescriptionField.js]: Could you please add a comment before each empty div returned referencing this PR, so the reader can know about why we're doing this? Something like `// See #312: Ensure compatibility with old version of React.`

File: package.json (hunk 2 above)
Comment (rom10) [inline package.json]: 👍

File: src/components/fields/DescriptionField.js (hunk 3 above)
Comment (rom10) [inline src/components/fields/DescriptionField.js]: Done

**Comment by n1k0** [comment]
Thank you :+1:

**Comment by n1k0** [comment]
Released in [v0.41.0](https://github.com/mozilla-services/react-jsonschema-form/releases/tag/v0.41.0).

**Pull request merged**